temporalio/bridge/src/worker.rs

```diff
@@ -204,3 +216,51 @@ impl TryFrom<WorkerConfig> for temporal_sdk_core::WorkerConfig {
             .map_err(|err| PyValueError::new_err(format!("Invalid worker config: {}", err)))
     }
 }
+
+/// For feeding histories into core during replay
+#[pyclass]
+pub struct HistoryPusher {
+    tx: Option<Sender<HistoryForReplay>>,
+}
+
+impl HistoryPusher {
+    fn new() -> (Self, ReceiverStream<HistoryForReplay>) {
+        let (tx, rx) = channel(1);
+        (Self { tx: Some(tx) }, ReceiverStream::new(rx))
+    }
+}
+
+#[pymethods]
+impl HistoryPusher {
+    fn push_history<'p>(
+        &self,
+        py: Python<'p>,
+        workflow_id: &str,
+        history_proto: &PyBytes,
+    ) -> PyResult<&'p PyAny> {
+        let history = History::decode(history_proto.as_bytes())
+            .map_err(|err| PyValueError::new_err(format!("Invalid proto: {}", err)))?;
+        let wfid = workflow_id.to_string();
+        let tx = if let Some(tx) = self.tx.as_ref() {
+            tx.clone()
+        } else {
+            return Err(PyRuntimeError::new_err(
+                "Replay worker is no longer accepting new histories",
+            ));
+        };
+        future_into_py(py, async move {
+            tx.send(HistoryForReplay::new(history, wfid))
+                .await
+                .map_err(|_| {
+                    PyRuntimeError::new_err(
+                        "Channel for history replay was dropped, this is an SDK bug.",
+                    )
+                })?;
+            Ok(())
+        })
+    }
+
+    fn close(&mut self) {
+        self.tx.take();
```

Comment:
I think being explicit is better here. It ensures we stop cramming in more histories as quickly as possible, and since we know when we're done it's easy to do.

Reply:
I guess so long as the user never sees it no prob.